### ISSUE TITLE
Delivery Issue after deleting around 200 dead accounts

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#6236]
### Expected behavior
When I delete a lot of accounts the delivery should still work as usual.

### Actual behavior
After deleting 200 dead accounts (accounts w/o login and accounts w/o any items and last login more than 8 month ago) my delivery queues exploded.
Before:  170 - 500 - 50 (estimated, from my memory)
After: 165 - 35135 - 11923 

The queues are on that high level for more than 12 hours now.

I understand that the deferred worker queue (in the middle) will stay high for some days until all deleted accounts are actually removed. But I think that the worker queue (the right one) should not stay at that high level. It should go down step by step. But it doesn't.

And the problem is delivery stopped working. No delivery at all. I cannot send posts to the forums and public posts don't arrive on other nodes (Friendica, Diaspora, Hubzilla, Mastodon). I also tried to post to my _local_ support forum about the problem. But my post never arrived there.

I have set 25 workers in admin panel. And I don't have any errors in my PHP log about memory issues or something similar.

Before I deleted those accounts everything was working.

Anything I can look for in the logs?

### Steps to reproduce the problem
Well, delete more than 200 accounts and watch what happens. ;-)

### Friendica version you encountered the problem
2018.12-dev (on current develop code)

### Friendica source (git, zip)
git

### PHP version
7.2.10

### SQL version
10.1.34 MariaDB

**Comment by AlfredSK** [comment]
Notice: The receiving part still works. I get posts and comments almost instantly.

**Comment by annando** [comment]
Please activate the debug log level and execute this in the shell:
```
tail -f /log/to/friendica.log | grep -a :Worker.php:.*Load
```
Please report here some of the lines (approx. 10) here. It should look nearly like this:
```
Load: 8.4/20 - processes: 4143/5/161 - jpm: 11/77/234 (0:1, 30:3/51, 40:2/114, 50:0/1) - maximum: 6/30
```

**Comment by AlfredSK** [comment]
I just posted something from my other node mordor.social (also on develop) to my support forum on libranet.de - it worked. So, it seems to be no code issue.

@annando Thanks. I will do that.

**Comment by annando** [comment]
BTW: Please set the ```system/worker_debug``` config to true for that.

**Comment by AlfredSK** [comment]
ok.
I had this `dbclean-expire-limit = 1000` set in my config. I have deleted that and added `worker_debug = true`.

Here is the output from the log:

````
2018-12-05T15:16:30Z@log:30305:5c07eac16c284149 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36642/8/10314 - jpm: 4/9/18 (0:1, 20:5/7958, 30:3/256, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:31Z@log:30305:5c07eac16c284149 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36636/8/10319 - jpm: 4/9/18 (0:1, 20:5/7957, 30:3/256, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:31Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36636/8/10318 - jpm: 4/9/18 (0:1, 20:5/7962, 30:3/256, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:31Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36637/8/10317 - jpm: 4/9/18 (0:1, 20:5/7961, 30:3/256, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:32Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36633/8/10324 - jpm: 4/9/18 (0:1, 20:5/7968, 30:3/256, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:32Z@log:32257:5c07ebb237073150 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36634/8/10324 - jpm: 5/9/18 (0:2, 20:4/7967, 30:2/255, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:33Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36633/8/10326 - jpm: 5/9/18 (0:1, 20:5/7971, 30:3/255, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:33Z@log:30305:5c07eac16c284149 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36635/8/10325 - jpm: 5/9/18 (0:1, 20:5/7970, 30:3/255, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:33Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.84/10 - processes: 36635/8/10324 - jpm: 5/9/18 (0:1, 20:5/7969, 30:3/255, 40:0/2099, 50:0/9) - maximum: 9/35
2018-12-05T15:16:34Z@log:32152:5c07eba0b0128122 [Debug]:Worker.php:707:tooMuchWorkers   Load: 3.69/10 - processes: 36630/8/10325 - jpm: 5/9/18 (0:1, 20:5/7970, 30:3/255, 40:0/2099, 50:0/9) - maximum: 9/35
````

**Comment by AlfredSK** [comment]
The load setting is confusing. I thought a load of 10 is already very high on a 6-core system.

**Comment by AlfredSK** [comment]
Ok. I tried a load setting of 20, 30 and 50. It can handle the 50. The actual load is 5.x - which is ok-ish for the 6 cores. And I went back to my original setting of max. 25 workers + fastlane.

````
2018-12-05T15:49:01Z@log:18235:5c07f33ee0114193 [Debug]:Worker.php:707:tooMuchWorkers   Load: 5.21/50 - processes: 38192/18/8316 - jpm: 18/30/18 (0:5, 20:7/6447, 30:1/1, 40:4/1871, 50:0/9) - maximum: 18/25
````

**Comment by AlfredSK** [comment]
jpm: 57/42/22 now

**Comment by annando** [comment]
Hmm ... look here:
```Load: 3.94/20 - processes: 4236/16/215 - jpm: 559/291/259 (0:1, 30:11/201, 40:5/30) - maximum: 16/30```

That's weird how slow your system is. Have you got optimal SQL settings?

Please also execute ```optimize table workerqueue;``` and have a look if the values are higher. The value after the ```0:``` indicates how much processes are unassigned. They are waiting for the next task. This value should be around 1 or 2, not 5.

**Comment by AlfredSK** [comment]
Yes. Database works well and fast. Mysqltuner doesn't complain.
The queues are usually pretty short.
And I had never values for jpm as high as yours. Maybe it depends on the actual tasks the system has to process.

**Comment by AlfredSK** [comment]
My tables are not fragmented, mysqltuner reports. And I optimized the workerqueue already.

**Comment by AlfredSK** [comment]
jpm looks better now: jpm: 192/257/216

**Comment by AlfredSK** [comment]
> The value after the 0: indicates how much processes are unassigned. They are waiting for the next task. This value should be around 1 or 2, not 5.

This value after the 0: goes down to 1. But then it also goes up to 25 or 30 for some time. At the moment it is oscillating around 18-20.

**Comment by AlfredSK** [comment]
@annando  Just for the record: `jpm: 320/229/165 (0:1, ...`
And I didn't change anything.

**Comment by MrPetovan** [comment]
If you had a bunch of long-running tasks like post deletions, it could have explained the low jpm number. This is what #6237 addressed.
